### PR TITLE
[APPSEC-786] Specify top secrets related to Google

### PR DESCRIPTION
**Pull request by jamie-anderson-sonarsource**
https://sonarsource.atlassian.net/browse/APPSEC-786

* Adding support for Google OAuth 2.0.
* Minor tweaks for Google API keys.

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          # "secret"
+          # any text, containing up to 1 newline
+          # the target client secret
+          # any text, containing up to 3 newlines
+          # ".apps.googleusercontent.com"
+          pattern: "(?i)(?:secret(?:[^\\r\\n]*+\\r?\\n)?[^\\r\\n]+)\\b([A-Za-z0-9_\\-]{24})\\b(?:(?:[^\\r\\n]*+\\r?\\n){0,3}[^\\r\\n]*\\.apps\\.googleusercontent\\.com\\b)"
+        post:
+          patternNot: "(?i)sample|example|x{6}|0{6}"
+          statisticalFilter:
+            threshold: 4.0
```
Comment (gaetan-ferry-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml:147]: Same thing here. 3.8 might be a more accurate threshold.
```suggestion
            threshold: 3.8
```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      metadata:
+        name: Google OAuth client secrets should not be disclosed
+      detection:
+        matching:
+          # "secret"
+          # any text, containing up to 1 newline
+          # the target client secret
+          # any text, containing up to 3 newlines
+          # ".apps.googleusercontent.com"
+          pattern: "(?i)(?:secret(?:[^\\r\\n]*+\\r?\\n)?[^\\r\\n]+)\\b([A-Za-z0-9_\\-]{24})\\b(?:(?:[^\\r\\n]*+\\r?\\n){0,3}[^\\r\\n]*\\.apps\\.googleusercontent\\.com\\b)"
```
Comment (gaetan-ferry-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml:143]: This one pattern is a bit afraying. It matches quite a few false positives on sourcegraph. Hopefully the statistical filter will do its job here.

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM with a suggestion to lower the entropy threshold a bit.

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml (hunk 2 above)
Comment (jamie-anderson-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/google-oauth2.yaml]: I've adjusted this regex to try and avoid as many false positives as possible. This makes the regex a lot harder to read, but it does help.

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**